schema-engine/sql-schema-describer/src/postgres.rs

```diff
@@ -1122,6 +1150,79 @@ impl<'a> SqlSchemaDescriber<'a> {
         Ok(())
     }
 
+    /// Return the constraints that are not primary keys, foreign keys, or unique keys.
+    /// Namely, this returns CHECK and EXCLUDE constraints.
+    async fn get_constraints(
+        &self,
+        sql_schema: &mut SqlSchema,
+    ) -> DescriberResult<IndexMap<(String, String), BitFlags<TableProperties, u8>>> {
```

Comment:
I don't think we need the `IndexMap` at all here: if you use the table id for the same purpose as `constraint_key` in this function, you can mutate the `TableProperties` for the table inside the schema directly. That would be less code and avoids the memory churn with IndexMap. And indexing a `Vec` is cheaper than hash lookups an `IndexMap`.